incubator/hnc/pkg/kubectl/create2.go

```diff
@@ -0,0 +1,52 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+var create2Cmd = &cobra.Command{
+	Use:   "create2 -n parent child1 [child2 ...]",
+	Short: "Creates subnamespaces under the given parent.",
+	Args:  cobra.RangeArgs(1, 10),
+	Run: func(cmd *cobra.Command, args []string) {
+		parent, _ := cmd.Flags().GetString("namespace")
+		if parent == "" {
+			fmt.Println("Error: parent must be set via --namespace or -n")
+			os.Exit(1)
+		}
+		// Create hns instances, the custom resources representing the self-serve subnamespaces.
+		// TODO: ensure the specified children don't already exist and the parent is allowed
+		//  to create subnamespaces through the admission controller. See issue
+		//  https://github.com/kubernetes-sigs/multi-tenancy/issues/458
+		createHNSs(client, parent, args)
+	},
+}
+
+func createHNSs(cl Client, nnm string, hnnms []string) {
+	for _, hnnm := range hnnms {
+		cl.createHierarchicalNamespace(nnm, hnnm)
+	}
+}
+
+func newCreate2Cmd() *cobra.Command {
+	create2Cmd.Flags().StringP("namespace", "n", "", "The parent namespace for the new child namespace(s)")
```

Comment:
nit: Maybe we need to unify the name for subnamespaces? In line 27, it is "subnamespaces", here it is "child namesoace(s)", and line 35 it is "self-serve subnamespaces" (may not be a concern since it is a comment). Maybe we change "child namespace(s)" here to "subnamespaces" too? Also I assume subnamespaces == self-serve subnamespaces?

Reply:
SG! I will change all to "self-serve subnamespaces" since it's the most accurate.